sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -380,6 +380,9 @@ object ColumnPruning extends Rule[LogicalPlan] {
         p
       }
 
+    // Eliminate the Projects with empty projectList
+    case p @ Project(projectList, child) if projectList.isEmpty => child
```

Comment:
Let me respond the original question by @cloud-fan 
We will not see an empty Project, if the child has more than one columns. The empty Project only happens after columnPruning. I am fine, if we want to add an extra rule for eliminating Project only.